Telemetry/TelemetryMeasurements.swift

```diff
@@ -126,37 +126,47 @@ public class DistributionMeasurement: StaticTelemetryMeasurement {
 public class EventsMeasurement: TelemetryMeasurement {
     private let storage: TelemetryStorage
     private let pingType: String
-    
-    private var events: [[Any?]]
-    
+    private var eventsAddedToFile = 0
+
+    // Ensure at least this many events have been added before uploading.
     public var numberOfEvents: Int {
         get {
-            return events.count
+            return eventsAddedToFile
         }
     }
     
     init(storage: TelemetryStorage, pingType: String) {
         self.storage = storage
         self.pingType = pingType
-        
-        self.events = storage.get(valueFor: "\(pingType)-events") as? [[Any?]] ?? []
```

Comment:
You are going to need to read in the number of events you already have on disk here at init. Otherwise, you'll be incorrectly starting your count at `0` even if you already have some written to disk.

To get this starting count efficiently, you may want to add another function to `TelemetryStorage` called `numberOfEvents(forPingType pingType: String)` and change your delimiter from `,` to `,\n`. This way, you can simply read in the number of newlines in the file at startup to get your number.